optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverFactory.java

```diff
@@ -26,68 +26,60 @@
 import org.optaplanner.core.api.domain.solution.PlanningSolution;
 import org.optaplanner.core.config.SolverConfigContext;
 import org.optaplanner.core.config.solver.SolverConfig;
-import org.optaplanner.core.impl.solver.EmptySolverFactory;
-import org.optaplanner.core.impl.solver.XStreamXmlSolverFactory;
+import org.optaplanner.core.config.solver.SolverConfigs;
+import org.optaplanner.core.impl.solver.DefaultSolverFactory;
 
 /**
- * Builds {@link Solver} instances.
+ * Creates {@link Solver} instances.
+ * Most applications only need one SolverFactory.
  * <p>
- * To build an instance, use {@link #createFromXmlResource(String)} or any of the other creation methods.
- * <p>
- * Supports tweaking the configuration programmatically before a {@link Solver} instance is build.
+ * To build an instance, use {@link #createFromXmlResource(String)}.
+ * To change the configuration programmatically, create a {@link SolverConfig} with {@link SolverConfigs}
+ * and use {@link #create(SolverConfig)}.
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
 public abstract class SolverFactory<Solution_> {
```

Comment:
I moved the KieContainer and Emtpy methods down in this class, making the diff a mess...